flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/BigIntComparator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.typeutils.base;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.MemorySegment;
+
+@Internal
+public final class BigIntComparator extends BasicTypeComparator<BigInteger> {
+
+	private static final long serialVersionUID = 1L;
+
+	public BigIntComparator(boolean ascending) {
+		super(ascending);
+	}
+
+	@Override
+	public int compareSerialized(DataInputView firstSource, DataInputView secondSource) throws IOException {
+		BigInteger bi1 = BigIntSerializer.readBigInteger(firstSource);
+		BigInteger bi2 = BigIntSerializer.readBigInteger(secondSource);
+		int comp = bi1.compareTo(bi2);
```

Comment:
This is intentional. The Comparator does not support `null` comparisions. It is the same behavior than `StringComparator` has: serialize null, but do not compare it or even check it.